### ISSUE TITLE
Setting CoreWebView2NewWindowRequestedEventArgs.NewWindow to null throws a NullReferenceException

### ISSUE THREAD

**Issue by ztanczos** [MicrosoftEdge/WebView2Feedback#2412]
**Description**
I have a WPF Window where I added a WebView2 control:
```
<Grid>
  <wv:WebView2 x:Name="_wvControl" Source="https://www.microsoft.com" />
</Grid>
```
If I subscribe to the NewWindowRequested event and in the event handler I set NewWindow to null I get a NullReferenceException:
```
        public MainWindow()
        {
            InitializeComponent();
            _wvControl.CoreWebView2InitializationCompleted += _wvControl_CoreWebView2InitializationCompleted;
        }

        private void _wvControl_CoreWebView2InitializationCompleted(object sender, Microsoft.Web.WebView2.Core.CoreWebView2InitializationCompletedEventArgs e)
        {
            _wvControl.CoreWebView2.NewWindowRequested += CoreWebView2_NewWindowRequested;
        }

        private void CoreWebView2_NewWindowRequested(object sender, Microsoft.Web.WebView2.Core.CoreWebView2NewWindowRequestedEventArgs e)
        {
            e.NewWindow = null;
        }
```

Stacktrace:
   at Microsoft.Web.WebView2.Core.CoreWebView2NewWindowRequestedEventArgs.set_NewWindow(CoreWebView2 value)
   at WebViewTestBed.MainWindow.CoreWebView2_NewWindowRequested(Object sender, CoreWebView2NewWindowRequestedEventArgs e) in C:\Users\ztanc\source\repos\WebViewTestBed\WebViewTestBed\MainWindow.xaml.cs:line 36

**Version**
Version: 1.0.1185.39
Framework: WPF, .NET Framework 4.7.2
OS: Windows 11

[AB#39388345](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/39388345)

**Comment by champnic** [comment]
Hey @ztanczos - I believe this is By Design. The NewWindow is the WebView2 that will be handling the New Window request, so null isn't valid. What is the behavior you are looking to achieve?

**Comment by ztanczos** [comment]
Hi @champnic 
I believe null is a valid value because by default e.NewWindow is null but even in that case if I set it to null again (i.e.: null = null), I get the exception.
The above example is not realistic, it's just meant to illustrate the problem. A more realistic example would be:
e.NewWindow = ConstructNewWindow();
where that function would either create a new window or return null if no new window is needed depending on some business logic.

If setting NewWindow to null is invalid then I'd expect ArgumentNullException instead. The guidelines say that public APIs shouldn't throw nullref.

**Comment by champnic** [comment]
That's a good point, thanks for the extra info! I've added this bug to our backlog. Just want to confirm that this isn't blocking your work, as you should be able to have a check and only set the value if it's non-null.

**Comment by ztanczos** [comment]
No, this isn't blocking us. Thanks for looking into it!

**Comment by champnic** [comment]
This is fixed in runtimes 102.0.1248.0+
